CONTRIBUTING.md

```diff
@@ -1,10 +1,113 @@
-#Contributing
+# Contributing
+When contributing to this repository, please first discuss the change you wish to make via issue, email, or any other method with the owners of this repository before making a change.
+Please note we have a code of conduct, please follow it in all your interactions with the project.
+[Code of Conduct](#code-of-conduct) 
 
-For contributing please follow the instructions below:
-1. Fork the repository on GitHub
-2. Clone the project to your own machine
-3. Commit changes to your own branch
-4. push your work back up to your forked repo
-5. Submit a Pull Request so that we can review your changes
+# Pull Request Process
+1. Create Issue with relevant information about your PR
+2. Fork the repository on GitHub
+3. Clone the project to your own machine
+4. create a branch with relevent name for your pr
+5. Commit changes to your own branch
+6. push your work back up to your forked repo
+7. Submit a Pull Request so that we can review your changes
+8. Link the PR to the relevant issue
+9. Send the PR to the slack group and ask for reviewer
+10.  Ensure that your code follows the PEP 8 Style guide for Python code before submitting a pull request.
+    We have also set up Flake8 to ensure that our style guides are being followed(PR's that dont pass this may not get reviewed until those are solved)
 
-NOTE: Be sure to fetch and merge the latest changes from the "upstream" repository before making a pull request!
+**Note: Be sure to fetch and merge the latest changes from the "upstream" repository before making a pull request!**
+
+**Note: If your PR do not meet one of the requirements, the PR will not be reviewed!**
+
+#### Pull request template:
+1. Making changes/Creating XFILE
+2. Empty line
+3. What?
+4. Why?
+5. How?
+6. Testing?
+7. Screenshots(if needed)
+8. Anything else?
+
+### Commiting:
+•   This project is following [commit convention](https://www.conventionalcommits.org/en/v1.0.0/) and it has to be followed through
```

Comment:
If this is the commit convention, the entire team should be up to it. 
Did you decide about it together?

Reply:
I think this commit convention is a bit overkill for this project because we won't use any automated tools on top our commits.
I will change it to something that suits better to are needs.